src/sparsezoo/analysis/analysis.py

```diff
@@ -0,0 +1,439 @@
+                for node_analysis in node_analyses
+                if node_analysis.outputs and "float" in node_analysis.outputs[0].dtype
+            ]
+        )
+        num_dense_floating_point_ops = sum(
+            [
+                node_analysis.num_dense_ops
+                for node_analysis in node_analyses
+                if node_analysis.outputs and "float" in node_analysis.outputs[0].dtype
+            ]
+        )
+
+        return cls(
+            layer_counts=layer_counts,
+            non_parameterized_operator_counts=op_counts,
+            num_dense_ops=sum(
+                [node_analysis.num_dense_ops for node_analysis in node_analyses]
+            ),
+            num_sparse_ops=sum(
+                [node_analysis.num_sparse_ops for node_analysis in node_analyses]
+            ),
+            num_sparse_layers=len(
+                [
+                    None
+                    for node_analysis in node_analyses
+                    if node_analysis.is_sparse_layer
+                ]
+            ),
+            num_quantized_layers=len(
+                [
+                    None
+                    for node_analysis in node_analyses
+                    if node_analysis.is_quantized_layer
+                ]
+            ),
+            num_parameters=num_parameters,
+            num_sparse_parameters=num_sparse_parameters,
+            num_four_blocks=num_four_blocks,
+            num_sparse_four_blocks=num_sparse_four_blocks,
+            num_dense_quantized_ops=num_dense_quantized_ops,
+            num_sparse_quantized_ops=num_sparse_quantized_ops,
+            num_sparse_floating_point_ops=num_sparse_floating_point_ops,
+            num_dense_floating_point_ops=num_dense_floating_point_ops,
+            average_sparsity=(num_sparse_parameters / num_parameters),
+            average_four_block_sparsity=(num_sparse_four_blocks / num_four_blocks),
+            nodes=node_analyses,
+        )
+
+    @staticmethod
+    def analyze_nodes(model_onnx: ModelProto) -> List[NodeAnalysis]:
+        """
+        :param: model that contains the nodes to be analyzed
+        :return: list of node analyses from model graph
+        """
+        node_shapes, node_dtypes = extract_node_shapes_and_dtypes(model_onnx)
+
+        nodes = []
+        for node in model_onnx.graph.node:
+            node_analysis = NodeAnalysis.from_node(
+                model_onnx, node, node_shapes=node_shapes, node_dtypes=node_dtypes
+            )
+            nodes.append(node_analysis)
+
+        return nodes
+
+    def yaml(self, file_path: Optional[str] = None) -> Union[str, None]:
+        """
+        :param file_path: optional file path to save yaml to
+        :return: if file_path is not given, the state of this analysis model
+            as a yaml string, otherwise None
+        """
+        if file_path is None:
+            return yaml.dump(self.dict(), allow_unicode=True, sort_keys=False)
```

Comment:
YMMV, but I'd make one call to `yaml.dump` and use a local variable... then handle the variable differently rather than repeat the `yaml.dump` call